src/featurestorebundle/widgets/WidgetsFactory.py

```diff
@@ -37,19 +40,13 @@ def create_for_target(self):
         self.__widgets.add_text("number_of_time_units", self.__defaults.number_of_time_units)
 
     def create_target_name(self):
+        targets = [
+            row.__getattr__(get_target_id_column_name())
+            for row in self.__targets_reader.read_enum().select(get_target_id_column_name()).collect()
```

Comment:
does this run every time a notebook is run?

Reply:
even every time you you changes widgets but its pretty fast as the table is small